### PR TITLE
pkgsMusl.colord: fix build

### PR DESCRIPTION
**Pull request by alyssais**
###### Description of changes

We can't build the daemon in pkgsMusl, because [rustc does not support dynamic Musl targets][1], which means we can't build Spidermonkey, which means we can't build Polkit.  But it still makes sense to support the rest of colord, because an application should still be able to link against it to talk to a colord daemon built in some other way (e.g. provided by a different distro).

[1]: https://github.com/NixOS/nixpkgs/issues/179242

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/colord/default.nix [hunk 1]
```diff
@@ -18,14 +18,15 @@
...
-, wrapGAppsHook
+, wrapGAppsNoGuiHook
 , shared-mime-info
 , sane-backends
 , docbook_xsl
 , docbook_xsl_ns
 , docbook_xml_dtd_412
 , gtk-doc
 , libxslt
+, enableDaemon ? !stdenv.hostPlatform.isMusl || stdenv.hostPlatform.isStatic
```
Comment (jtojnar) [inline pkgs/tools/misc/colord/default.nix:29]: Does it work on static, or should it be

```suggestion
, enableDaemon ? !(stdenv.hostPlatform.isMusl || stdenv.hostPlatform.isStatic)
```

Comment (alyssais) [reply pkgs/tools/misc/colord/default.nix:29]: It works on static — in fact the problem is that rustc assumes that "musl" always means static, and that's why it doesn't work with dynamic musl.

**Comment by alyssais** [comment]
Doesn't build of OfBorg due to having recently acquired a transitive dependency of gperftools, which is currently broken on Musl.  Will be fixed by #179102.

File: pkgs/tools/misc/colord/default.nix [hunk 2]
```diff
@@ -83,11 +85,10 @@ stdenv.mkDerivation rec {
     gusb
     lcms2
     libgudev
-    polkit
     sane-backends
     sqlite
     systemd
-  ];
+  ] ++ lib.optional enableDaemon polkit;
```
[the remark this answers is not in the document]
Comment (alyssais) [reply pkgs/tools/misc/colord/default.nix]: I preferred it as it was, but I've made the change.

**Review by jtojnar — APPROVED** [review]

**Comment by alyssais** [comment]
_Sigh._ OfBorg lib tests have failed due to a curl test failure. Let's try again.

@OfBorg eval

**Pull request merged**